### DISCUSSION TITLE
Authentication event fires every time a tab is refocused (NextJS)

### DISCUSSION THREAD

**Discussion by react-ghost** [supabase/supabase#7690] [Questions] [answered]
Utilizing Next.Js with supabase, everytime I refocus the tab my Auth Listener fires a new 'SIGNED_IN' event type.

What this was causing for me was unexpected re-directs that I couldn't figure out due to my useEffect in a React Context being set up like the following:

```
useEffect(() => {
    setAuth(supabase.auth.session());

    const { data: authListener } = supabase.auth.onAuthStateChange(
      async (event, session) => {
        handleAuthChange(event, session);
        if (event === "SIGNED_OUT") {
          resetAllState();
          router.push("/");
          return;
        }

        if (event === "SIGNED_IN") {
          alert('Sign In Event');
          setAuth(session);
          setUser(session.user);
          await checkUser();
          router.push("/hosting/dashboard");
          return;
        }
        if (event === "TOKEN_REFRESHED") {
          setAuth(session);
          setUser(session.user);
          await checkUser();
          router.push("/hosting/dashboard");
        }
      }
    );

    checkUser();
    return () => {
      authListener.unsubscribe();
    };
  }, []);
```

The reason this causes a problem is because rather than being able to wait for my user data to be fetched before I redirect the user to their dashboard, I am forced to manually set a timeout which gives the react context the time to set a user state which is what allows the redirect to occur without the user being pushed back to the home page because it thinks there is not a user yet.

Think of it like

```
useEffect(() => {
   if(!userContext.user) {
       router.push("/");
      }
}, [])
```

The convenience of doing this when the right 'event' was triggered was being able to wait for my user data to be populated.

Does anyone know why this is happening?

**Comment by GaryAustin1** [comment, marked as answer]
This is due to a recent change in gotrue-js to keep token's refreshed reliably when in background or sleeping devices...
https://github.com/supabase/supabase/issues/7250

**Reply by react-ghost** [reply]
I see. I know this is not the repo for anything web development related, but do you happen to know how I should handle this since I can't rely on the auth listener to communicate only the true actual sign in event now? Is there a way to filter these?

**Reply by react-ghost** [reply]
Nevermind, found example in thread you linked - thanks!
